src/ipaddr.coffee

```diff
@@ -381,7 +381,7 @@ expandIPv6 = (string, parts) ->
     return null
 
   # Remove zone index and save it for later
-  zoneId = (string.match(zoneIndex) || [])[0]
+  zoneId = (string.match(ipv6Regexes['zoneIndex']) || [])[0]
```

Comment:
So `zoneIndex` has always been a string:

```coffee
zoneIndex = "%[0-9a-z]{1,}"
```

That was then referenced by multiple other RegExp definitions:

```coffee
ipv6Regexes =
   zoneIndex:    new RegExp zoneIndex, 'i'
   native:       new RegExp "^(::)?(#{ipv6Part})?([0-9a-f]+)?(::)?(#{zoneIndex})?$", 'i'
   transitional: new RegExp "^((?:#{ipv6Part})|(?:::)(?:#{ipv6Part})?)" +
                            "#{ipv4Part}\\.#{ipv4Part}\\.#{ipv4Part}\\.#{ipv4Part}" +
                            "(#{zoneIndex})?$", 'i'
```

The trouble is that this particular changed line (Line 384) was accidentally incorrectly using the string `zoneIndex` rather than the RegExp `ipv6Regexes['zoneIndex']` as it was clearly intending to.

The funny part is that it was accidentally working in most cases because of the strange defaulting behavior of `String.prototype.match`: while it expects 1 RegExp parameter, it will [implicitly handle a non-RegExp argument by wrapping it with the `RegExp` constructor](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#A_non-RegExp_object_as_the_parameter). In our case, that means it was working like `new RegExp zoneIndex` instead of the intended `new RegExp zoneIndex, 'i'`.